src/ServiceWorkerCacher.ts

```diff
@@ -1,36 +1,54 @@
 import Cacher from "./Cacher";
 import Store from "./Store";
 import Manifest from "./Manifest";
+import * as HTMLUtilities from "./HTMLUtilities";
+import ApplicationCacheCacher from "./ApplicationCacheCacher";
 
-/** Class that caches responses using ServiceWorker's Cache API. */
+const template = `
+    <div class="cache-status"></div>
+`;
+
+/** Class that caches responses using ServiceWorker's Cache API, and optionally
+    falls back to the application cache if service workers aren't available. */
 export default class ServiceWorkerCacher implements Cacher {
     private readonly serviceWorkerPath: string;
     private readonly store: Store;
+    private readonly manifestUrl: URL;
     private readonly areServiceWorkersSupported: boolean;
+    private readonly fallbackCacher: ApplicationCacheCacher | null;
+    private statusElement: HTMLDivElement;
+    private cachingStarted: boolean = false;
+    private cachingFinished: boolean = false;
```

Comment:
`cachingStarted` and `cachingFinished` can probably be collapsed into a single member of an enum type (e.g. `NotYetStarted`, `Started`, and `Finished`) so we don't have to worry about maintaining some kind of `cachingFinished == true` ==> `cachingStarted == true` invariant.